smart-contracts/contracts/mixins/MixinFunds.sol

```diff
@@ -19,11 +19,11 @@ contract MixinFunds
     address _tokenAddress
   ) public
   {
+    tokenAddress = _tokenAddress;
```

Comment:
Isn't that introducing a re-entrancy challenge? if the target contracts does wonky stuff on `totalSupply`?

Reply:
The recommendation from Slither was to change the state (i.e. set tokenAddress) before making the external call to totalSupply just in case someone attempts to reenter from the totalSupply function.  So this should reduce risk if anything.

That said making this change is overkill.  New with Solidity 0.5 (vs 0.4.*) was static call support.  That actually blocks reentrancy attacks here.